operator/pkg/values/translation/enricher.go

```diff
@@ -0,0 +1,35 @@
+package translation
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/operator-framework/helm-operator-plugins/pkg/values"
+	"github.com/stackrox/rox/pkg/k8sutil"
+	"helm.sh/helm/v3/pkg/chartutil"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// Enricher reads helm values produced for an object by a Translator or another enricher, and returns new values.
+// Note: mutates provided vals.
+type Enricher interface {
+	Enrich(ctx context.Context, obj k8sutil.Object, vals chartutil.Values) (chartutil.Values, error)
+}
+
+// WithEnrichment chains a given translator with zero or more enrichers.
+func WithEnrichment(translator values.Translator, enrichers ...Enricher) values.Translator {
```

Comment:
nit: could also be implemented as options pattern, but needs an update in our fork. Likely to be too much effort and annoying but wanted to leave it here.
See https://golang.cafe/blog/golang-functional-options-pattern.html in "Option 3: Functional Options Pattern". Feel free to ditch this.

Reply:
This is a great idea, but I'll defer this because I've ventured far already in scope of ROX-9156.

Reply:
https://issues.redhat.com/browse/ROX-21426